packages/metro/src/DeltaBundler/__tests__/traverseDependencies-test.js

```diff
@@ -681,6 +693,408 @@ describe('edge cases', () => {
+     * │ /bundle │ ──▶ │ /foo │ ──▶ │ /bar │
+     * └─────────┘     └──────┘     └──────┘
+     *                   │
+     *                   │
+     *                   ▼
+     *                 ┌──────┐
+     *                 │ /baz │
+     *                 └──────┘
+     *
+     * and end up with the following graph:
+     *
+     * ┌─────────┐     ┌───────┐
+     * │ /bundle │ ──▶ │ /quux │
+     * └─────────┘     └───────┘
+     *
+     * In the process, we add a "live" edge from /bundle to /quux (which makes it to the final
+     * graph) and a "dead" edge from /foo to /quux (which does not, because /foo is pruned).
+     *
+     * The order and chunking of mutations should not affect the correctness of the graph
+     * (but has been known to cause bugs, hence these regression tests).
+     */
+    it('all in one delta, adding the live edge first', async () => {
+      await initialTraverseDependencies(graph, options);
+
+      /*
+      ┌─────────┐     ┌──────┐     ┌──────┐
+      │ /bundle │ ──▶ │ /foo │ ──▶ │ /bar │
+      └─────────┘     └──────┘     └──────┘
+                        │
+                        │
+                        ▼
+      ┏━━━━━━━━━┓     ┌──────┐
+      ┃  /quux  ┃     │ /baz │
+      ┗━━━━━━━━━┛     └──────┘
+      */
+      Actions.createFile('/quux');
+
+      /*
+      ┌─────────┐     ┌──────┐     ┌──────┐
+      │ /bundle │ ──▶ │ /foo │ ──▶ │ /bar │
+      └─────────┘     └──────┘     └──────┘
+        ┃	              │
+        ┃               │
+        ▼               ▼
+      ┌─────────┐     ┌──────┐
+      │  /quux  │     │ /baz │
+      └─────────┘     └──────┘
+      */
+      Actions.addDependency('/bundle', '/quux');
+
+      /*
+      ┏━━━━━━━━━━━━━━━━━━━━┓
+      ┃                    ┃
+      ┃  ┌─────────┐     ┌──────┐     ┌──────┐
+      ┃  │ /bundle │ ──▶ │ /foo │ ──▶ │ /bar │
+      ┃  └─────────┘     └──────┘     └──────┘
+      ┃    │               │
+      ┃    │               │
+      ┃    ▼               ▼
+      ┃  ┌─────────┐     ┌──────┐
+      ┗▶ │  /quux  │     │ /baz │
+         └─────────┘     └──────┘
+      */
+      Actions.addDependency('/foo', '/quux');
+
+      /*
+      ┌────────────────────┐
+      │                    │
+      │  ┌─────────┐   / ┌──────┐     ┌──────┐
+      │  │ /bundle │ ┈/▷ │ /foo │ ──▶ │ /bar │
+      │  └─────────┘ /   └──────┘     └──────┘
+      │    │               │
+      │    │               │
+      │    ▼               ▼
+      │  ┌─────────┐     ┌──────┐
+      └▶ │  /quux  │     │ /baz │
+         └─────────┘     └──────┘
+      */
+      Actions.removeDependency('/bundle', '/foo');
+
+      /*
+      Compute the delta for the current graph:
+      ┌─────────┐     ┌───────┐
+      │ /bundle │ ──▶ │ /quux │
+      └─────────┘     └───────┘
+      (modified)       (added)
+      */
+      expect(
+        getPaths(await traverseDependencies([...files], graph, options)),
+      ).toEqual({
+        added: new Set(['/quux']),
```

Comment:
I'm guessing [the PR](https://github.com/facebook/metro/pull/506) didn't fix the appropriate delta calculations